efel/settings.py

```diff
@@ -116,9 +116,12 @@ def set_setting(self,
         if hasattr(self, setting_name):
             expected_types = {f.name: f.type for f in fields(self)}
             expected_type = expected_types.get(setting_name)
-            if expected_type and not isinstance(new_value, expected_type):
-                raise ValueError(f"Invalid value for setting '{setting_name}'. "
-                                 f"Expected type: {expected_type.__name__}.")
+            if expected_type is not None:
+                if expected_type == float and isinstance(new_value, int):
```

Comment:
Are we sure that we would only have int to float case here? Maybe we could be more general by having a try/except in which we just try to convert the variable

Reply:
Yes I thought about this but then it will for example allow boolean values for int features, do we want that?

Reply:
Indeed, allowing boolean to int might be a bit of a stretch. But I think that we should at least allow also **float to int** (if the value does not change), that might happen a lot, and **int to bool**, we have this in multiple legacy scripts. Also we might have other edge cases, like numpy.int64 that are not taken care of here and that we should allow.
I think it is better to allow too much (e.g. bool to int) with a try / except, than making efel crash on a lot of edge cases we did not anticipate. We could have an additional check afterwards (e.g. `assert value_before == value_after_conversion`) to be sure that we are not screwing the values (e.g. float to int case with `int(1.1)` )
What do you think?

Reply:
I agree, I will make the changes, we could even provide a warning message to the user when we make a conversion

Reply:
Nice change! About the message above, I don't think we have to make a warning every time we do a conversion.
But when we change the value (e.g. input=1.1, int(input) -> 1), then I think we should either make a warning or raise an error, given that the input is far from what was expected. What do you think?

Reply:
Indeed, after some consideration I have opted for debug statements. I agree, it seems prudent to make a warning in that case, I will add it

Reply:
I don't think we should always make a warning for float -> int. When the value does not change, like converting 1.0 to 1, I don't think we should throw a warning. It seems like something that is expected and would occur often.

Reply:
good point, I have changed it so it raises a warning only when truncation happens